### PR TITLE
[Graphbolt]Add data format

### PR DESCRIPTION
**Pull request by peizhou001**
## Description
Add 2 types of new data format.

## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [ ] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

File: python/dgl/graphbolt/data_format.py [hunk 1]
```diff
@@ -11,14 +11,26 @@ class LinkPredictionEdgeFormat(Enum):
...
+    TAILCONDITIONED: Represents the 'conditioned' format where data is
+    structured as triples `[u, v, [negative tails]]`, where  '(u, v)' signifies
+    the source and destination nodes of positive edges, while 'u' collaborates
+    with each node in '[negative tails]' to create negative edges.
     """
 
-    CONDITIONED = "conditioned"
     INDEPENDENT = "independent"
+    CONDITIONED = "conditioned"
+    HEADCONDITIONED = "headconditioned"
```
Comment (Rhett-Ying) [inline python/dgl/graphbolt/data_format.py:35]: `HEAD_CONDITIONED`?

Comment (frozenbugs) [reply python/dgl/graphbolt/data_format.py:35]: +1

File: python/dgl/graphbolt/data_format.py [hunk 2]
```diff
@@ -11,14 +11,26 @@ class LinkPredictionEdgeFormat(Enum):
     in link prediction:
 
     Attributes:
+    INDEPENDENT: Represents the 'independent' format where data is structured
+    as triples `[u, v, label]` indicating the source and destination nodes of
+    an edge, with a label (0 or 1) denoting it as negative or positive.
+
     CONDITIONED: Represents the 'conditioned' format where data is
     structured as quadruples `[u, v, [negative heads], [negative tails]]`
```
Comment (frozenbugs) [inline python/dgl/graphbolt/data_format.py:19]: Worth to callout the [negative heads], [negative tails] should have the same length.

File: python/dgl/graphbolt/data_format.py [hunk 3]
```diff
@@ -11,14 +11,26 @@ class LinkPredictionEdgeFormat(Enum):
...
 
-    INDEPENDENT: Represents the 'independent' format where data is structured
-    as triples `[u, v, label]` indicating the source and destination nodes of
-    an edge, with a label (0 or 1) denoting it as negative or positive.
+    HEAD_CONDITIONED: Represents the 'head conditioned' format where data is
+    structured as triples `[u, v, [negative heads]]`, where  '(u, v)' signifies
+    the source and destination nodes of positive edges, while each node in
+    '[negative heads]' collaborates with 'v' to create negative edges.
+
+    TAIL_CONDITIONED: Represents the 'conditioned' format where data is
```
Comment (frozenbugs) [inline python/dgl/graphbolt/data_format.py:27]:
```suggestion
    TAIL_CONDITIONED: Represents the 'tail conditioned' format where data is
```

**Comment by frozenbugs** [comment]
Please also add unit test for this 2 new types.

**Review by frozenbugs — APPROVED** [review]

**Pull request merged**